config.yaml

```diff
@@ -0,0 +1,35 @@
+linkerdVersion: stable-2.8.0
+externalIssuer: false
+controlPlane:
+    # namespace: l5d-conformance
+    config:   
+        ha: false
+        flags:
+            - "--controller-log-level"
+            - "debug"
+            - "--proxy-log-level"
+            - "warn,linkerd2_proxy=debug"
+      # addOns:
+      #     tracing:
+      #         enabled: true
+      #         collector:
+      #             resources:
+      #                 cpu:
+      #                     limit: 100m
+      #                     request: 10m
+      #                 memory:
+      #                     limit: 100Mi
+      #                     request: 50Mi
+lifecycle:
+    skip: false
```

Comment:
Instead of specifying `skip: [false|true]` for each test case, what do you think of utilizing regex to pattern match test functions to be skipped?

E.g.,

```yaml
testSuite:
   skip: ["TestIngress.*", "^MyFlakyTest$"]
```

Reply:
Utiizing regex patterns would be useful if these could be passed down to the `--skip` flag of the [Ginkgo CLI](https://onsi.github.io/ginkgo/#the-ginkgo-cli). But in our case, we're not programatically determining the values for any of the CLI flags, and neither does Ginkgo natively provide APIs to skip tests on the basis of regex patterns.
To programatically determine if a Ginkgo test should be skipped would mean having to check a condition under the `ginkgo.BeforeEach` block and calling `ginkgo.Skip`. I don't see any advantage over matching regex patterns in the code than simply determining boolean values to evaluate an `if` condition. Moreover, we also have to consider that our users will not be familiar with the name of the test functions in our code (like `TestIngress`, etc) when they choose to run our Sonobuoy plugin - they only have to deal with setting up a correct config YAML and running the plugin. We could document these names, but I'm not really sure if this makes our user experience better.

Ginkgo also works little differently from the standard `testing` library. Each of our test suites, actually has only one `Test*` function, which holds several [Ginkgo 'containers'](https://onsi.github.io/ginkgo/#structuring-your-specs). For example, the ingress test suite contains only one function `TestIngress` that holds our Ginkgo specs, and similarly for inject (TestInject), etc. Using regex for such similar names seems a little superfluous to me. 

If the redundancy of the `skip` flag is an issue, what do you think about having to skip on the basis of test names rather than regex pattern? For example:
```yaml
testSuite:
  skip:
    - ingress
    - inject
    - myFlakyTest
```

Reply:
Can we just pass the regex to the `go test` (adding a `-run` flag) in the `bin/go-test`? (I assume that's the entrypoint to run the test.) Feels like a lot of boiler plate code to have to do this for every spec.

```
	ginkgo.BeforeEach(func() {
		if c.SkipIngress() {
			ginkgo.Skip("Skipping ingress tests")
		}
	})
```

Reply:
Hmm, but this might not work too well when running with the Sonobuoy workflow where the scripts are hidden away from the users and all they have to do is defined the configuration in the YAML and mount it as a configMap.
The config YAML is supposed to be the single source for reading any kind of configuration. What do you think?

Reply:
For now, I've made this boilerplate a single reusable function :
```go
func ShouldTestSkip(skip bool, message string) bool {
	return ginkgo.BeforeEach(func() {
		if skip {
			ginkgo.Skip(message)
		}
	})
}
```

This helper is called in the beginning of every skippable test:
```go
var _ = ginkgo.Describe("`linkerd inject`", func() {
	_, config := utils.GetHelperAndConfig()
	_ = utils.ShouldTestSkip(config.SkipInject(), "Skipping inject tests")
        
       // remaining specs go here
})

I hope this solves temporarily solves the problem of redundant boilerplate code. :smile: